src/segment_kubectl.go

```diff
@@ -31,15 +51,80 @@ func (k *kubectl) init(props *properties, env environmentInfo) {
 }
 
 func (k *kubectl) enabled() bool {
+	parseKubeConfig := k.props.getBool(ParseKubeConfig, false)
+	if parseKubeConfig {
+		return k.doParseKubeConfig()
+	} else {
+		return k.doCallKubectl()
+	}
+}
+
+func (k *kubectl) doParseKubeConfig() bool {
+	// Follow kubectl search rules (see https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)
+	// TL;DR: KUBECONFIG can contain a list of files. If it's empty ~/.kube/config is used. First file in list wins when merging keys.
+	kubeconfigs := filepath.SplitList(k.env.getenv("KUBECONFIG"))
+	if len(kubeconfigs) == 0 {
+		kubeconfigs = []string{filepath.Join(k.env.homeDir(), ".kube/config")}
+	}
+	contexts := make(map[string]KubeConfigContext)
+	k.Context = ""
+	for _, kubeconfig := range kubeconfigs {
+		if kubeconfig == "" {
+			continue
+		}
+
+		content := k.env.getFileContent(kubeconfig)
+
+		var config KubeConfig
+		err := yaml.Unmarshal([]byte(content), &config)
+		if err != nil {
+			continue
+		}
+
+		for _, context := range config.Contexts {
+			if _, exists := contexts[context.Name]; !exists {
+				contexts[context.Name] = context.Context
+			}
+		}
+
+		if k.Context == "" {
+			k.Context = config.CurrentContext
+		}
+
+		context, exists := contexts[k.Context]
+		if exists {
+			k.KubeConfigContext = context
+			return true
+		}
+	}
+
+	displayError := k.props.getBool(DisplayError, false)
+	if displayError {
+		message := "KUBECONFIG ERR"
+		if k.Context == "" {
+			k.Context = message
+		}
+		k.Namespace = message
+		k.User = message
+		k.Cluster = message
+		return true
+	} else {
```

Comment:
the else can be removed due to returning early in the if. I would swap them in that case to remove the need for the indentation:

```golang
if !displayError {
    return false
}
message := "KUBECONFIG ERR"
if k.Context == "" {
	k.Context = message
}
k.Namespace = message
k.User = message
k.Cluster = message
return true
```

Reply:
Done